solver/llbsolver/solver.go

```diff
@@ -178,42 +192,45 @@ func (s *Solver) Solve(ctx context.Context, id string, req frontend.SolveRequest
 					}
 					m[k] = workerRef.ImmutableRef
 
-					dt, err := inlineCache(ctx, exp.CacheExporter, res)
-					if err != nil {
-						return nil, err
-					}
-					if dt != nil {
-						inp.Metadata[fmt.Sprintf("%s/%s", exptypes.ExporterInlineCache, k)] = dt
+					if inlineCacheExporter != nil {
+						dt, err := inlineCache(ctx, inlineCacheExporter, res)
+						if err != nil {
+							return nil, err
+						}
+						if dt != nil {
+							inp.Metadata[fmt.Sprintf("%s/%s", exptypes.ExporterInlineCache, k)] = dt
+						}
 					}
 				}
 			}
 			inp.Refs = m
 		}
 
-		if err := inVertexContext(j.Context(ctx), e.Name(), "", func(ctx context.Context) error {
-			exporterResponse, err = e.Export(ctx, inp)
+		if err := inVertexContext(j.Context(ctx), exporterInstance.Name(), "", func(ctx context.Context) error {
+			exporterResponse, err = exporterInstance.Export(ctx, inp)
 			return err
 		}); err != nil {
 			return nil, err
 		}
 	}
 
-	var cacheExporterResponse map[string]string
-	if e := exp.CacheExporter; e != nil {
+	var cacheExporterResponses []map[string]string
+	for _, cer := range cacheExporterRequests {
```

Comment:
Does this export inline twice? Probably the same in old version as well but not sure if I'm missing something.

Reply:
Possibly this could run concurrently. Could we add a test that uses registry and local together (or all three) as inline is already a special case with different code-path?

Reply:
manifest list without manifests was generated 🤔 https://github.com/moby/buildkit/pull/829#issuecomment-543559435

Reply:
sorry, could you show the entire `buildctl build` flags that should be tested

Reply:
In https://github.com/moby/buildkit/pull/829/files#diff-fce84029d346504e44323c30d21a3808R1825 add `type=registry` as well. Or an additional case where`local` and `registry` are used together.